apps/webapp/app/header/navbar.tsx

```diff
@@ -1,45 +1,51 @@
 'use client';
 
 import Link from 'next/link';
-import { usePathname } from 'next/navigation';
 import { DappPath } from './paths';
 import { cn } from '@penumbra-zone/ui/lib/utils';
+import { useTypesafePathname } from '../../hooks/typesafe-pathname';
 
 const links = [
```

Comment:
Please create an interface for this. Also, make `subLinks` an optional field so you don't need to do `subLinks: []` a lot.